jscomp/runtime/caml_obj.res

```diff
@@ -303,53 +299,46 @@ type eq = (Obj.t, Obj.t) => bool
     basic type is not the same, it will not equal 
 */
 let rec equal = (a: Obj.t, b: Obj.t): bool =>
-  /* front and formoest, we do not compare function values */
   if a === b {
     true
   } else {
     let a_type = Js.typeof(a)
-    if (
-      a_type == "string" ||
-        (a_type == "number" ||
-        (a_type == "bigint" ||
-          (a_type == "boolean" ||
-          (a_type == "undefined" || a === %raw(`null`)))))
-    ) {
+    if a_type !== "object" || a === %raw(`null`) {
       false
     } else {
       let b_type = Js.typeof(b)
-      if a_type == "function" || b_type == "function" {
-        raise(Invalid_argument("equal: functional value"))
-      } /* first, check using reference equality */
```

Comment:
Since we checked the functions using strict equal `a === b`, it doesn't make sense to raise an error at this point.

Reply:
Why not?
The error says you cannot compare functional values.
We might change the stance on this, but both choices are valid design choices.

Reply:
I think according to the funciton name `equal` it makes sense that when you compare two non-identical function values, it returns `false`. Also, it improves performance for non `function` values, since there's no need to specifically check for it.